forms/HeaderField.php

```diff
@@ -84,4 +70,30 @@ public function getAttributes() {
 	public function Type() {
 		return null;
 	}
+
+	/**
+	 * Header fields support dynamic titles via schema state
+	 *
+	 * @return array
+	 */
+	public function getSchemaStateDefaults() {
+		$state = parent::getSchemaStateDefaults();
+
+		$state['data']['title'] = $this->Title();
+
+		return $state;
+	}
+
+	/**
+	 * Header fields heading level to be set
+	 *
+	 * @return array
+	 */
+	public function getSchemaDataDefaults() {
+		$data = parent::getSchemaDataDefaults();
+
+		$data['headingLevel'] = $this->headingLevel;
```

Comment:
Properties specific to individual fields should probably be in $data[`data`][someprop]